MIGRATING-0.6-0.7.md

```diff
@@ -396,4 +288,259 @@ Note that the table above shows the new fonts' `ascii` variants only. Some new f
+
+### Monospace fonts
+
+Monospaced fonts should now be built by creatign a `MonoFont` struct and assigning it to a `const`. The following example shows a migration of a font with 5x9 pixel characters using the Latin-1 encoding.
+
+```diff
+// The font bitmap has 32 character glyphs per row.
+const CHARS_PER_ROW: u32 = 32;
+
+// Each character is 5x9 px.
+const GLYPH_SIZE: Size = Size::new(5, 9);
+
+- // Map a given character to an index in the glyph bitmap
+- fn char_offset_impl(c: char) -> u32 {
+-     let fallback = '?' as u32 - ' ' as u32;
+-     if c < ' ' {
+-         return fallback;
+-     }
+-     if c <= '~' {
+-         return c as u32 - ' ' as u32;
+-     }
+-     if c < '\u{00A0}' || c > 'ÿ' {
+-         return fallback;
+-     }
+-     c as u32 - ' ' as u32 - 33
+- }
+-
+- #[derive(Debug, Copy, Clone)]
+- pub struct ExampleFont {}
+- impl Font for ExampleFont {
+-     const FONT_IMAGE: &'static [u8] = include_bytes!("../data/ExampleFont.raw");
+-     const CHARACTER_SIZE: Size = Size::new(5, 9);
+-     const FONT_IMAGE_WIDTH: u32 = Self::CHARACTER_SIZE.width * CHARS_PER_ROW;
+-
+-     fn char_offset(c: char) -> u32 {
+-         char_offset_impl(c)
+-     }
+- }
++ use embedded_graphics::{
++     geometry::Size,
++     image::ImageRaw,
++     mono_font::{mapping::StrGlyphMapping, DecorationDimensions, MonoFont},
++ };
++
++ pub const EXAMPLE_FONT: MonoFont = MonoFont {
++     image: ImageRaw::new_binary(
++         // In this example, this equals 32 characters per row, each character 5px across.
++         160,
++     ),
++     // A glyph mapping with base ASCII range and Latin1 range.
++     glyph_mapping: &StrGlyphMapping::new("\0 ~\0\u{00A0}ÿ", '?' as usize - ' ' as usize),
++     character_size: Size::new(5, 9),
++     character_spacing: 0,
++     baseline: 7,
++     underline: DecorationDimensions::new(8, 1),
++     strikethrough: DecorationDimensions::new(4, 1),
++ };
+```
+
+If a more complex or dynamic glyph mapping is required, a function can be provided instead:
+
+```rust
+use embedded_graphics::{
+    geometry::Size,
+    image::ImageRaw,
+    mono_font::{mapping::StrGlyphMapping, DecorationDimensions, MonoFont},
+};
+
+pub const EXAMPLE_FONT: MonoFont = MonoFont {
+    image: ImageRaw::new_binary(
+        include_bytes!("../data/ExampleFont.raw"),
+        // In this example, this equals 32 characters per row, each character 5px across.
+        160,
+    ),
+    glyph_mapping: &char_offset,
+    character_size: Size::new(5, 9),
+    character_spacing: 0,
+    baseline: 7,
+    underline: DecorationDimensions::new(8, 1),
+    strikethrough: DecorationDimensions::new(4, 1),
+};
+
+fn char_offset(c: char) -> usize {
+    let fallback = '?' as usize - ' ' as usize;
+    if c < ' ' {
+        return fallback;
+    }
+    if c <= '~' {
+        return c as usize - ' ' as usize;
+    }
+    if c < '\u{00A0}' || c > 'ÿ' {
+        return fallback;
+    }
+    c as usize - ' ' as usize - 33
+}
+```
```

Comment:
Now that the example above doesn't use `StrGlyphMapping` anymore I would suggest to use a single example to demonstrate a string mapping and using a function. I've also changed the mapping to one that isn't included as a predefined mapping.

```suggestion
Custom mappings between characters and glyph positions can be used by using `StrGlyphMapping`, using a function or implementing the `GlyphMapping` trait:

```rust
use embedded_graphics::{
    geometry::Size,
    image::ImageRaw,
    mono_font::{mapping::StrGlyphMapping, DecorationDimensions, MonoFont},
};

pub const EXAMPLE_FONT: MonoFont = MonoFont {
    image: ImageRaw::new_binary(
        include_bytes!("../data/digits.raw"),
        // In this example, this equals 32 characters per row, each character 15px across.
        10 * 15,
    ),
    glyph_mapping: &StrGlyphMapping::new("\009", 0),
    // or use a function:
    // glyph_mapping: &digit_mapping,
    character_size: Size::new(15, 30),
    character_spacing: 5,
    baseline: 29,
    underline: DecorationDimensions::default_underline(30),
    strikethrough: DecorationDimensions::default_strikethrough(30),
};

fn digit_mapping(c: char) -> usize {
    if c >= '0' || c <= '9' {
        c as usize - '0' as usize
    } else {
        0
    }
}
```